tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SNS/CachedMessageHeadersHelper.cs

```diff
@@ -41,11 +42,11 @@ static CachedMessageHeadersHelper()
 
             messageAttributeIL.Emit(OpCodes.Dup);
             messageAttributeIL.Emit(OpCodes.Ldarg_0);
-            messageAttributeIL.Emit(OpCodes.Callvirt, messageAttributeValueType.GetProperty("StringValue").GetSetMethod());
+            messageAttributeIL.Emit(OpCodes.Callvirt, messageAttributeValueType.GetProperty("BinaryValue").GetSetMethod());
 
             messageAttributeIL.Emit(OpCodes.Ret);
 
-            _createMessageAttributeValue = (Func<string, object>)createMessageAttributeValueMethod.CreateDelegate(typeof(Func<string, object>));
+            _createMessageAttributeValue = (Func<MemoryStream, object>)createMessageAttributeValueMethod.CreateDelegate(typeof(Func<MemoryStream, object>));
```

Comment:
Was there a reason why we didn't we use duck-typing here to access the `DataType` and `BinaryValue` properties?

I realize this happened in a previous PR, so no need to fix it in this now if it works, but maybe something to fix later? (assuming it's even possible to use duck-typing here)

Reply:
Inside this dynamic method we are creating a new MessageAttributeValue, pushing it onto the stack, and then calling methods on it...theoretically we could use ducktyping, but then we'd have to insert calls to our DuckTyping API's and the existing approach seems a lot simpler